modin/pandas/groupby.py

```diff
@@ -263,7 +268,9 @@ def nth(self, n, dropna=None):
         return self._default_to_pandas(lambda df: df.nth(n, dropna=dropna))
 
     def cumsum(self, axis=0, *args, **kwargs):
-        result = self._apply_agg_function(lambda df: df.cumsum(axis, *args, **kwargs))
+        result = self._apply_agg_function(
+            lambda df: df.cumsum(axis, *args, **kwargs), selection=self._selection
```

Comment:
doing `selection=self._selection` bypasses the deselection of the categorical columns, in the general case we want them to be deselected, but there are `cum*` and `nunique` functions which don't do deselection

Reply:
`_apply_agg_function` and `_wrap_aggregation` now takes the `selection` parameter, if it's specified it passes the selection to the backend as-is, if not, it applies the general rule that it doesn't want to see categorical columns in the selection. [[link]](https://github.com/YarShev/modin/blob/25e8460fa5665fd9240d1a0c96574cacb1092836/modin/pandas/groupby.py#L1109-L1111)